Serialiser_Engine/Objects/MemberMapConventions/ImmutableTypeClassMapConventionFixed.cs

```diff
@@ -89,11 +90,18 @@ public void Apply(BsonClassMap classMap)
                 // if any constructors were mapped by this convention then map all the properties also
                 foreach (var property in properties)
                 {
-                    var memberMap = classMap.MapMember(property);
-                    if (classMap.IsAnonymous)
+                    try
                     {
-                        var defaultValue = memberMap.DefaultValue;
-                        memberMap.SetDefaultValue(defaultValue);
+                        var memberMap = classMap.MapMember(property);
```

Comment:
Not against a try-catch here in general, but good to understand the reasoning behind why it was required now? Sorry if I have missed an explanation somewhere.

Reply:
Because I managed to make the code crash when I initially created an Enumeration that was not inheriting from `IObject`. 
Although it doesn't crash anymore, I felt it was a good idea to wrap an area of the code that had the potential to cause a crash into a try-catch.

Reply:
FYI, I have raised an issue to look into more details into the representation of an Enumeration in Json as well as the process of its serialisation itself. 
https://github.com/BHoM/BHoM_Engine/issues/2696